src/date/expand-pattern.js

```diff
@@ -1,3 +1,24 @@
+function getDateTime( type, cldr ) {
+	var result;
+	result = cldr.main([
+		"dates/calendars/gregorian/dateTimeFormats",
+		type
+	]);
+	if ( result ) {
+		result = formatMessage( result, [
+			cldr.main([
+				"dates/calendars/gregorian/timeFormats",
+				type
+			]),
+			cldr.main([
+				"dates/calendars/gregorian/dateFormats",
+				type
+			])
+		]);
+	}
+	return result;
+}
+
```

Comment:
1. By placing this function outside the `define` scope, it's outside the module and therefore won't built correctly.
1. Given you have created this function, why aren't you using it on `case "datetime" in pattern:` below? You have duplicate block of code.